datadog-transformer.rb

```diff
@@ -1,21 +1,41 @@
 require 'net/http'
 require 'yaml'
+require 'json'
 
 class JSONTransformer
   def initialize
     @config = File.exists?("/opt/nginx_dogstats.yaml") ? YAML.load_file("/opt/nginx_dogstats.yaml") : {}
     @gce_zone = self.resolve_gce_zone
+    @aws_region = self.resolve_aws_region
   end
 
+  # https://cloud.google.com/compute/docs/storing-retrieving-metadata#querying
   def resolve_gce_zone
     uri = URI("http://metadata.google.internal/computeMetadata/v1/instance/zone")
     req = Net::HTTP::Get.new(uri)
     req['Metadata-Flavor'] = 'Google'
     begin
-      resp =  Net::HTTP.start(uri.hostname, uri.port) { |http| http.request(req) }
+      resp = Net::HTTP.start(uri.hostname, uri.port, open_timeout: 1, read_timeout: 1) { |http|
```

Comment:
open and read timeouts both default to 60 seconds in ruby std lib. this optimizes for cases where this wouldn't resolve